clients/src/main/java/org/apache/kafka/common/Cluster.java

```diff
@@ -151,7 +151,10 @@ public PartitionInfo partition(TopicPartition topicPartition) {
      * @return A list of partitions
      */
     public List<PartitionInfo> partitionsForTopic(String topic) {
-        return this.partitionsByTopic.get(topic);
+        List<PartitionInfo> parts = this.partitionsByTopic.get(topic);
+        if (parts == null)
```

Comment:
Now I see why it wasn't compiling, the outer <List> is not needed, just <PartitionInfo> type is enough.
Fixed, thanks for the tip.